### PR TITLE
VP9e - use C++ primitives (memcpy => std::copy)

### PR DESCRIPTION
**Pull request by ilapichev**
Issue: MDP-50976

### CODE REVIEW

File: _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_ddi.h [hunk 1]
```diff
@@ -185,22 +185,27 @@ typedef struct tagENCODE_CAPS_VP9
...
+        std::copy(std::begin(ivf_file_header),std::end(ivf_file_header), pBitstream);
+
+        return MFX_ERR_NONE;
     };
 
     inline void AddPictureHeader(mfxU8* pBitstream, mfxU32 bufferSize)
     {
         mfxU32 ivf_frame_header[3] = { 0x00000000, 0x00000000, 0x00000000 };
-        memcpy_s(pBitstream, bufferSize, ivf_frame_header, sizeof (ivf_frame_header));
+        std::copy(std::begin(ivf_frame_header),std::end(ivf_frame_header), pBitstream);
```
Comment (lakulako) [inline _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_ddi.h]: std::fill_n(begin, 12, 0) better?

File: _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp [hunk 2]
```diff
@@ -75,7 +75,7 @@ mfxStatus MFXVideoENCODEVP9_HW::Query(mfxCoreInterface *core, mfxVideoParam *in,
                     {
                         mfxExtBuffer *pCorrectedBuf = GetExtendedBuffer(toValidate.ExtParam, toValidate.NumExtParam, pInBuf->BufferId);
                         MFX_CHECK_NULL_PTR1(pCorrectedBuf);
-                        memcpy_s(pOutBuf, pOutBuf->BufferSz, pCorrectedBuf, pCorrectedBuf->BufferSz);
+                        *pOutBuf = *pCorrectedBuf;
```
Comment (lakulako) [inline _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp]: It copies only buffer header, but not complete buffer!

File: _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp [hunk 3]
```diff
@@ -903,7 +903,7 @@ mfxStatus MFXVideoENCODEVP9_HW::GetVideoParam(mfxVideoParam *par)
         mfxExtBuffer *pLocalBuf = GetExtendedBuffer(m_video.ExtParam, m_video.NumExtParam, pOutBuf->BufferId);
         if (pLocalBuf && (pOutBuf->BufferSz == pLocalBuf->BufferSz))
         {
-            memcpy_s(pOutBuf, pOutBuf->BufferSz, pLocalBuf, pLocalBuf->BufferSz);
+            *pOutBuf = *pLocalBuf;
```
Comment (lakulako) [inline _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp]: same

File: _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp (hunk 2 above)
Comment (ilapichev) [reply _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp]: Yes, thanks, my merge error

File: _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_ddi.h (hunk 1 above)
Comment (ilapichev) [reply _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_ddi.h]: yeah, but the task is h265e- use C++ primitives (memcpy => std::copy)
Is it fine to change memcpy => std::fill_n?

File: _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp (hunk 2 above)
Comment (ilapichev) [reply _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp]: I had a closer look at this comment.
Not sure what do u mean?

Comment (ilapichev) [reply _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp]: std::copy(pCorrectedBuf, pCorrectedBuf + pCorrectedBuf->BufferSz, pOutBuf);

Comment (mgonchar) [reply _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp]: you need to cast it to byte pointers first, then you can do std::copy

**Pull request closed without merging**